cartridges/int_turnto_core_v5/cartridge/scripts/jobsteps/ExportCancelledOrders.js

```diff
@@ -0,0 +1,107 @@
+ */
+
+var Site = require('dw/system/Site');
+var File = require('dw/io/File');
+var FileWriter = require('dw/io/FileWriter');
+var Order = require('dw/order/Order');
+var OrderMgr = require('dw/order/OrderMgr');
+var Calendar = require('dw/util/Calendar');
+var Status = require('dw/system/Status');
+
+/* Script Modules*/
+var TurnToHelper = require('*/cartridge/scripts/util/TurnToHelperUtil');
+var OrderWriterHelper = require('*/cartridge/scripts/util/OrderWriterHelper');
+
+/**
+ * @function
+ * @name run
+ * @description The main function.
+ * @returns {dw.system.Status} The exit status for the job step
+ */
+exports.Run = function () {
+    var fileWriter = null;
+    try {
+        var args = arguments[0];
+
+        if (args.IsDisabled) {
+            return new Status(Status.OK, 'OK', 'Step disabled, skip it...');
+        }
+
+        // Load input Parameters
+        var exportFileName = args.ExportFileName;
+
+        // Test mandatory parameters
+        if (empty(exportFileName)) {
+            return new Status(Status.ERROR, 'ERROR', 'One or more mandatory parameters are missing. Export File Name = (' + exportFileName + ')');
+        }
+
+        var cancelledOrderDays = Site.getCurrent().getCustomPreferenceValue('turntoCancelledOrderDays');
+
+        if (empty(cancelledOrderDays)) {
+            return new Status(Status.ERROR, 'ERROR', 'Mandatory site preference "turntoHistoricalOrderDays" is missing. Export File Name = (' + exportFileName + ')');
+        }
+
+        // Get the file path where the output will be stored
+        var impexPath = File.getRootDirectory(File.IMPEX).getFullPath();
+
+        // loop through all allowed locales per site
+        TurnToHelper.getAllowedLocales().forEach(function (currentLocale) {
+            try {
+                var dateLimit = new Calendar();
+                dateLimit.add(Calendar.DAY_OF_YEAR, cancelledOrderDays * -1);
+
+                var query = 'lastModified >= {0} AND customerLocaleID = {1} AND (status = {2} OR status = {3})';
+                var orders = OrderMgr.searchOrders(query, 'creationDate asc', dateLimit.getTime(), currentLocale, Order.ORDER_STATUS_CANCELLED, Order.ORDER_STATUS_FAILED);
+
+                if (orders.count) {
+                    // Create a TurnTo directory if one doesn't already exist
+                    var turntoDir = new File(impexPath + File.SEPARATOR + 'TurnTo' + File.SEPARATOR + currentLocale);
+                    if (!turntoDir.exists()) {
+                        turntoDir.mkdirs();
+                    }
+
+                    // Initialize a file writer for output
+                    var orderExportFile = new File(turntoDir.getFullPath() + File.SEPARATOR + exportFileName + '_' + currentLocale + '_' + Site.getCurrent().ID + '.txt');
+
+                    fileWriter = new FileWriter(orderExportFile);
+
+                    fileWriter.writeLine('ORDERID\tSKU');
```

Comment:
This appears to be the header for the feed file.  It appears the `OrderWriterHelper` call below writes out a log more fields that just order id and sku.  How does this work?

Reply:
I created a second method `writeCancelledOrderData` in `OrderWriterHelper` that only writes the two fields. The original method `writeOrderData` writes all the order fields.

Reply:
Thanks for the clarification.  I was looking at the `OrderWriterHelper` on master, totally missing the one in the PR :facepalm: